File: FRONTEND/lib/COMMON/Core/Models/Emergence Issue Of The Month/emergence_issue_of_the_month_model.dart
import 'package:sudan_horizon_scanner/imports.dart';

List<IssueOfMonthModel> emergenceIssueOfTheMonthModelFromJson(String str) =>
    List<IssueOfMonthModel>.from(
        json.decode(str).map((x) => IssueOfMonthModel.fromJson(x)));

String emergenceIssueOfTheMonthModelToJson(List<IssueOfMonthModel> data) =>
    json.encode(List<dynamic>.from(data.map((x) => x.toJson())));

class IssueOfMonthModel {
  String? emergingIssue;
  double? repetition;
  int? totalDataCount;
  int? positiveSentimentAnalysisDataCount;
  int? neutralSentimentAnalysisDataCount;
  int? negativeSentimentAnalysisDataCount;
  String? time;
  String? summary;
  String? sdgTargeted;
  List<String>? sources;

  IssueOfMonthModel({
    this.emergingIssue,
    this.repetition,
    this.totalDataCount,
    this.positiveSentimentAnalysisDataCount,
    this.neutralSentimentAnalysisDataCount,
    this.negativeSentimentAnalysisDataCount,
    this.summary,
    this.time,
    this.sdgTargeted,
    this.sources,
  });

  factory IssueOfMonthModel.fromJson(Map<String, dynamic> json) =>
      IssueOfMonthModel(
        emergingIssue: json["emergingIssue"],
        repetition:
            json["repetition"] == null ? null : json["repetition"].toDouble(),
        totalDataCount: json["totalDataCount"] ?? null,
        positiveSentimentAnalysisDataCount:
            json["positiveSentimentAnalysisDataCount"] ?? null,
        neutralSentimentAnalysisDataCount:
            json["neutralSentimentAnalysisDataCount"] ?? null,
        negativeSentimentAnalysisDataCount:
            json["negativeSentimentAnalysisDataCount"] ?? null,
        sdgTargeted: json["sdgTargeted"] ?? null,
        summary: json["summary"] ?? null,
        time: json["time"] ?? null,
        sources: json["sources"] == null
            ? null
            : List<String>.from(json["sources"].map((x) => x)),
      );

  Map<String, dynamic> toJson() => {
        "emergingIssue": emergingIssue ?? null,
        "repetition": repetition ?? null,
    "totalDataCount": totalDataCount?? null,
        "positiveSentimentAnalysisDataCount":
            positiveSentimentAnalysisDataCount ?? null,
        "neutralSentimentAnalysisDataCount":
            neutralSentimentAnalysisDataCount ?? null,
        "negativeSentimentAnalysisDataCount":
            negativeSentimentAnalysisDataCount ?? null,
        "summary": summary ?? null,
        "time": time ?? null,
        "sdgTargeted": sdgTargeted ?? null,
        "sources":
            sources == null ? null : List<dynamic>.from(sources!.map((x) => x)),
      };
}

File: BACKEND/Models/Emergence Issue Of The Month/emergence_issue_of_the_month_model.js
// Import Mongoose
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

// Create Event Detail Schema and Model
const EmergenceIssueOfTheMonthSchema = new Schema({
    emergingIssue: {
        type: String
    },
    repetition: {
        type: Number
    },
    totalDataCount: { 
        type: Number
    },
    positiveSentimentAnalysisDataCount: {
        type: Number
    },
    neutralSentimentAnalysisDataCount: {
        type: Number
    },
    negativeSentimentAnalysisDataCount: {
        type: Number
    },
    time: {
        type: String
    },
    sdgTargeted: {
        type: String
    },
    summary: {
        type: String
    },
    sources: {
        type: [String]
    }
});


const EmergenceIssueOfTheMonthModel = mongoose.model('emergence_issue_of_the_month', EmergenceIssueOfTheMonthSchema, 'emergence_issue_of_the_month');
module.exports = {
    EmergenceIssueOfTheMonthSchema: EmergenceIssueOfTheMonthSchema,
    EmergenceIssueOfTheMonthModel: EmergenceIssueOfTheMonthModel
};